src/services/cherry-picker.ts

```diff
@@ -102,9 +46,23 @@ export class CherryPicker {
     for (const node of nodes) {
       rawNodes[node.publicKey] = node
       rawNodeIDs.push(node.publicKey)
-      const rawServiceLog = await this.fetchRawServiceLog(blockchain, node.publicKey)
+    }
+
+    // Pull all service & failure & error logs
+    const rawServiceLogs = await this.fetchRawLogs(blockchain, rawNodeIDs, 'service')
+    const rawFailureLogs = await this.fetchRawLogs(blockchain, rawNodeIDs, 'failure')
+    const rawErrorLogs = await this.fetchRawLogs(blockchain, rawNodeIDs, 'errors')
 
-      sortedLogs.push(await this.createUnsortedLog(node.publicKey, blockchain, rawServiceLog!))
+    for (const node of nodes) {
```

Comment:
Does order/sequential processing matter here? Wonder if we can also optimize this by also mapping over these with a `Promise.all`

Reply:
order does not matter at that point. would be a good code optimization after, but not messing w/ it now.